lighthouse-core/audits/seo/font-size.js

```diff
@@ -100,80 +100,51 @@ function getSelector(node) {
   return node.localName.toLowerCase();
 }
 
-/**
- * @param {FailingNodeData['node']} node
- * @return {{type: 'node', selector: string, snippet: string}}
- */
-function nodeToTableNode(node) {
-  const attributes = node.attributes || [];
-  const attributesString = attributes.map((value, idx) =>
-    (idx % 2 === 0) ? ` ${value}` : `="${value}"`
-  ).join('');
-
-  return {
-    type: 'node',
-    selector: node.parentNode ? getSelector(node.parentNode) : '',
-    snippet: `<${node.localName}${attributesString}>`,
-  };
-}
-
 /**
  * @param {string} baseURL
  * @param {FailingNodeData['cssRule']} styleDeclaration
  * @param {FailingNodeData['node']} node
- * @returns {{source: string, selector: string | {type: 'node', selector: string, snippet: string}}}
+ * @returns {{source: string | {sourceURL: string, url: string, line: number, column: number}, selector: string}}
  */
 function findStyleRuleSource(baseURL, styleDeclaration, node) {
-  if (
-    !styleDeclaration ||
-    styleDeclaration.type === 'Attributes' ||
-    styleDeclaration.type === 'Inline'
-  ) {
-    return {
-      selector: nodeToTableNode(node),
-      source: baseURL,
-    };
-  }
-
-  if (styleDeclaration.parentRule &&
+  if (styleDeclaration && styleDeclaration.parentRule &&
     styleDeclaration.parentRule.origin === 'user-agent') {
     return {
-      selector: styleDeclaration.parentRule.selectors.map(item => item.text).join(', '),
       source: 'User Agent Stylesheet',
+      selector: styleDeclaration.parentRule.selectors.map(item => item.text).join(', '),
     };
   }
 
-  if (styleDeclaration.type === 'Regular' && styleDeclaration.parentRule) {
```

Comment:
so we no longer require that it's `type: 'Regular'`. do we know what sort of impact this will have? what are cases it affects?

Reply:
`type` can be `Regular' | 'Inline' | 'Attributes'`

Regular is handled well here, so is Inline (I just use `styleDeclaration.stylesheet.isInline` where necessary instead). Attributes .... I should look into.

Reply:
At least, I _think_ `type = Inline` mean `style` tag? the comment on the old code L160-161 kinda suggests otherwise..

Reply:
Oh, inline here means via `<div style=`. attribute is ex: `<div width=`.

I'll add some cases for all these to my example page and see how to work with them.